cli/src/bin/phylum.rs

```diff
@@ -137,37 +167,8 @@ async fn handle_commands() -> CommandResult {
         }
     }
 
-    if matches.subcommand_matches("version").is_some() {
-        print_user_success!("{app_name} (Version {ver})");
-        return Ok(ExitCode::Ok.into());
-    }
-
-    if let Some(matches) = matches.subcommand_matches("update") {
-        let mut spinner = Spinner::new(
-            Spinners::Dots12,
-            "Downloading update and verifying binary signatures...".into(),
-        );
-        let res = update::do_update(matches.is_present("prerelease")).await;
-        spinner.stop_with_newline();
-        let message = res?;
-        print_user_success!("{}", message);
-        return Ok(ExitCode::Ok.into());
-    }
-
-    if let Some(matches) = matches.subcommand_matches("parse") {
-        return handle_parse(matches);
-    }
-
-    let timeout = matches
-        .value_of("timeout")
-        .and_then(|t| t.parse::<u64>().ok());
-
-    let ignore_certs =
-        matches.is_present("no-check-certificate") || config.ignore_certs.unwrap_or_default();
-    if ignore_certs {
-        log::warn!("Ignoring TLS server certificate verification per user request.");
-    }
-
+    // `auth` consumes `config` which prevents the API factory future from borrowing it, so we
+    // return early in order to guarantee that it has not been consumed after this point.
```

Comment:
We return early after every subcommand anyway, right? It's not like we wouldn't return if the `config` was borrowed?

Reply:
We do return after every command, but that "early" means "earlier than `PhylumApi` has a chance of borrowing it" (which would be, anywhere in the closing `match` block).

Reply:
`handle_submission()` also consumes `config`, but it gets to be in the `match` block... The difference, of course, is that (as we've discussed in another comment), the borrow ends when `api.await` occurs.

For `handle_auth()`, we don't need the API object, so the borrow can be ended with `drop(api)`. And then this can join the others in the `match` block.